docs/labels.md

```diff
@@ -11,50 +10,66 @@ key/value labels set on it, with at most one label with a particular key.
 }
 ```
 
-While there are no restrictions on the format of label values, label keys must be of the form:
-```
-label-key ::= prefixed-name | name
-prefixed-name ::= prefix '/' name
-prefix ::= DNS_SUBDOMAIN
-name ::= DNS_LABEL
-```
-DNS_LABEL and DNS_SUBDOMAIN are defined in the [identifiers design doc](/docs/design/identifiers.md). The prefix is optional. If the prefix is not specified, the key is assumed to be private to the user. Other system components that wish to use labels must specify a prefix. The "kubernetes.io/" prefix is reserved for use by kubernetes components.
+Unlike [names and UIDs](identifiers.md), labels do not provide uniqueness. In general, we expect many objects to carry the same label(s).
+Sets identified by labels could be overlapping (think Venn diagrams). For instance, a service might target all pods with `"tier": "frontend"` and  `"environment" : "prod"`.  Now say you have 10 replicated pods that make up this tier.  But you want to be able to 'canary' a new version of this component.  You could set up a `replicationController` (with `replicas` set to 9) for the bulk of the replicas with labels `"tier" : "frontend"` and `"environment" : "prod"` and `"track" : "stable"` and another `replicationController` (with `replicas` set to 1) for the canary with labels `"tier" : "frontend"` and  `"environment" : "prod"` and `"track" : canary`.  Now the service is covering both the canary and non-canary pods.  But you can mess with the `replicationControllers` separately to test things out, monitor the results, etc.
+
+Valid label keys are comprised of two segments - prefix and name - separated by a slash (`/`).  The name segment is required and must be a DNS label: 63 characters or less, all lowercase, beginning and ending with an alphanumeric character (`[a-z0-9]`), with dashes (`-`) and alphanumerics between.  The prefix and slash are optional.  If specified, the prefix must be a DNS
```

Comment:
Also, we extended the syntax to allow uppercase and underscores:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/util/validation.go#L46

Label values must be no more than 63 characters and are restricted to the same character set, but may be the empty string:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/util/validation.go#L33